packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,313 @@
+        }
+    }
+
+    private checkConstructorParameters(construct: ts.ConstructorDeclaration): void {
+        switch (this.options.mode) {
+            case Mode.Never:
+                for (const param of construct.parameters.filter(isParameterProperty))
+                    this.addFailureAtNode(
+                        param,
+                        FAILURE_STRINGS[Mode.Never],
+                        getFixerForDisallowedParameterProp(
+                            construct,
+                            param,
+                            getLineBreakStyle(this.context.sourceFile),
+                        ),
+                    );
+                break;
+            case Mode.WhenPossible:
+                for (const param of construct.parameters)
+                    if (!isParameterProperty(param) && canBeParameterProperty(param, construct))
+                        this.addFailureAtNode(
+                            param,
+                            FAILURE_STRINGS[Mode.WhenPossible],
+                            getFixerForLonghandProp(param, construct),
+                        );
+                break;
+            default:
+                if (construct.parameters.every(isParameterProperty)) return;
+
+                const allPropsCanBeParamProps = construct.parameters
+                    .filter((param) => !isParameterProperty(param))
+                    .every((param) => canBeParameterProperty(param, construct));
+
+                if (!allPropsCanBeParamProps && construct.parameters.some(isParameterProperty)) {
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            param.pos,
+                            param.end,
+                            FAILURE_STRINGS[Mode.Consistent].cannotBeParamPropsOnly,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+                } else if (allPropsCanBeParamProps && !construct.parameters.every(isParameterProperty)) {
+                    for (const param of construct.parameters.filter((p) => !isParameterProperty(p)))
+                        this.addFailure(
+                            param.pos,
+                            param.end,
+                            FAILURE_STRINGS[Mode.Consistent].canBeParamPropsOnly,
+                            getFixerForLonghandProp(param, construct),
+                        );
+                }
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+    const member = construct.parent!.members.find((mem) => isPropertyMatchForParam(mem, param));
+    if (member && !member.decorators && param.dotDotDotToken === undefined) {
```

Comment:
you can actually return early if `param.dotDotDotToken` is present or `param.name` is not an Identifier